No errors, unless file explicitly selected

This problem happens to me in the Diofant project (see all relevant configs [here](https://github.com/diofant/diofant/tree/b50066efea3974cb0249c6bed6cab3fae9dc37c2)).

Consider adding a trailing space:
```
$ git diff diofant/core/
diff --git a/diofant/core/expr.py b/diofant/core/expr.py
index c7857e0a5a..9fae5c3793 100644
--- a/diofant/core/expr.py
+++ b/diofant/core/expr.py
@@ -30,7 +30,7 @@ class Expr(Basic, EvalfMixin, metaclass=ManagedProperties):
 
     def __new__(cls, *args):
         obj = Basic.__new__(cls, *args)
-        obj._assumptions = cls.default_assumptions
+        obj._assumptions = cls.default_assumptions 
         return obj
 
     @property
```

Now run ``py.test`` with this file selected:
```

rootdir: /home/sk/src/diofant, inifile: setup.cfg
plugins: xdist-1.22.5, timeout-1.3.1, pylint-0.12.1, forked-0.2, cov-2.5.1, hypothesis-3.66.23
timeout: 600.0s
timeout method: signal
timeout func_only: False
collected 34 items / 33 deselected                                                                                                     
-----------------------------------------------------------------
Linting files
.
-----------------------------------------------------------------

diofant/core/expr.py F                                                                                                           [100%]

=============================================================== FAILURES ===============================================================
____________________________________________________ [pylint] diofant/core/expr.py _____________________________________________________
C: 33,50: Trailing whitespace (trailing-whitespace)
====================================================== slowest 30 test durations =======================================================
0.00s setup    diofant/core/expr.py
0.00s call     diofant/core/expr.py
0.00s teardown diofant/core/expr.py
```
Great, a problem detected.

But what if we run ``py.test`` without arguments or using directory instead of file?  Unfortunately, there will be no error in this case.  For example:
```
$ py.test -n0 --pylint -m pylint diofant/core
========================================================= test session starts ==========================================================
platform linux -- Python 3.5.3, pytest-3.7.1, py-1.5.4, pluggy-0.7.1

cache: True
ground types: gmpy

rootdir: /home/sk/src/diofant, inifile: setup.cfg
plugins: xdist-1.22.5, timeout-1.3.1, pylint-0.12.1, forked-0.2, cov-2.5.1, hypothesis-3.66.23
timeout: 600.0s
timeout method: signal
timeout func_only: False
collected 1489 items / 1422 deselected                                                                                                 
-----------------------------------------------------------------
Linting files
.
-----------------------------------------------------------------

diofant/core/__init__.py .                                                                                                       [  1%]
diofant/core/add.py .                                                                                                            [  2%]
diofant/core/alphabets.py .                                                                                                      [  4%]
diofant/core/assumptions.py .                                                                                                    [  5%]
diofant/core/basic.py .                                                                                                          [  7%]
diofant/core/cache.py .                                                                                                          [  8%]
diofant/core/compatibility.py .                                                                                                  [ 10%]
diofant/core/containers.py .                                                                                                     [ 11%]
diofant/core/coreerrors.py .                                                                                                     [ 13%]
diofant/core/decorators.py .                                                                                                     [ 14%]
diofant/core/evalf.py .                                                                                                          [ 16%]
diofant/core/evaluate.py .                                                                                                       [ 17%]
diofant/core/expr.py .                                                                                                           [ 19%]
diofant/core/exprtools.py .                                                                                                      [ 20%]
diofant/core/facts.py .                                                                                                          [ 22%]
diofant/core/function.py .                                                                                                       [ 23%]
diofant/core/logic.py .                                                                                                          [ 25%]
diofant/core/mod.py .                                                                                                            [ 26%]
diofant/core/mul.py .                                                                                                            [ 28%]
diofant/core/multidimensional.py .                                                                                               [ 29%]
diofant/core/numbers.py .                                                                                                        [ 31%]
diofant/core/operations.py .                                                                                                     [ 32%]
diofant/core/power.py .                                                                                                          [ 34%]
diofant/core/relational.py .                                                                                                     [ 35%]
diofant/core/rules.py .                                                                                                          [ 37%]
diofant/core/singleton.py .                                                                                                      [ 38%]
diofant/core/strategies.py .                                                                                                     [ 40%]
diofant/core/symbol.py .                                                                                                         [ 41%]
diofant/core/sympify.py .                                                                                                        [ 43%]
diofant/core/trace.py .                                                                                                          [ 44%]
diofant/core/tests/__init__.py .                                                                                                 [ 46%]
diofant/core/tests/test_args.py .                                                                                                [ 47%]
diofant/core/tests/test_arit.py .                                                                                                [ 49%]
diofant/core/tests/test_assumptions.py .                                                                                         [ 50%]
diofant/core/tests/test_basic.py .                                                                                               [ 52%]
diofant/core/tests/test_cache.py .                                                                                               [ 53%]
diofant/core/tests/test_compatibility.py .                                                                                       [ 55%]
diofant/core/tests/test_complex.py .                                                                                             [ 56%]
diofant/core/tests/test_containers.py .                                                                                          [ 58%]
diofant/core/tests/test_count_ops.py .                                                                                           [ 59%]
diofant/core/tests/test_diff.py .                                                                                                [ 61%]
diofant/core/tests/test_equal.py .                                                                                               [ 62%]
diofant/core/tests/test_eval.py .                                                                                                [ 64%]
diofant/core/tests/test_eval_power.py .                                                                                          [ 65%]
diofant/core/tests/test_evalf.py .                                                                                               [ 67%]
diofant/core/tests/test_evaluate.py .                                                                                            [ 68%]
diofant/core/tests/test_expand.py .                                                                                              [ 70%]
diofant/core/tests/test_expr.py .                                                                                                [ 71%]
diofant/core/tests/test_exprtools.py .                                                                                           [ 73%]
diofant/core/tests/test_facts.py .                                                                                               [ 74%]
diofant/core/tests/test_function.py .                                                                                            [ 76%]
diofant/core/tests/test_logic.py .                                                                                               [ 77%]
diofant/core/tests/test_match.py .                                                                                               [ 79%]
diofant/core/tests/test_multidimensional.py .                                                                                    [ 80%]
diofant/core/tests/test_noncommutative.py .                                                                                      [ 82%]
diofant/core/tests/test_numbers.py .                                                                                             [ 83%]
diofant/core/tests/test_operations.py .                                                                                          [ 85%]
diofant/core/tests/test_priority.py .                                                                                            [ 86%]
diofant/core/tests/test_relational.py .                                                                                          [ 88%]
diofant/core/tests/test_rules.py .                                                                                               [ 89%]
diofant/core/tests/test_strategies.py .                                                                                          [ 91%]
diofant/core/tests/test_subs.py .                                                                                                [ 92%]
diofant/core/tests/test_symbol.py .                                                                                              [ 94%]
diofant/core/tests/test_sympify.py .                                                                                             [ 95%]
diofant/core/tests/test_trace.py .                                                                                               [ 97%]
diofant/core/tests/test_truediv.py .                                                                                             [ 98%]
diofant/core/tests/test_var.py .                                                                                                 [100%]

====================================================== slowest 30 test durations =======================================================
0.01s call     diofant/core/evalf.py
0.00s teardown diofant/core/expr.py
0.00s teardown diofant/core/coreerrors.py
0.00s setup    diofant/core/alphabets.py
0.00s setup    diofant/core/tests/test_args.py
0.00s setup    diofant/core/expr.py
0.00s setup    diofant/core/coreerrors.py
0.00s setup    diofant/core/add.py
0.00s setup    diofant/core/__init__.py
0.00s setup    diofant/core/tests/test_equal.py
0.00s setup    diofant/core/tests/test_expand.py
0.00s call     diofant/core/strategies.py
0.00s teardown diofant/core/tests/test_subs.py
0.00s teardown diofant/core/tests/test_operations.py
0.00s call     diofant/core/tests/test_truediv.py
0.00s call     diofant/core/tests/test_exprtools.py
0.00s setup    diofant/core/tests/test_numbers.py
0.00s call     diofant/core/tests/test_eval_power.py
0.00s call     diofant/core/tests/test_multidimensional.py
0.00s setup    diofant/core/tests/test_evaluate.py
0.00s setup    diofant/core/tests/test_function.py
0.00s teardown diofant/core/exprtools.py
0.00s teardown diofant/core/logic.py
0.00s setup    diofant/core/compatibility.py
0.00s setup    diofant/core/tests/test_noncommutative.py
0.00s call     diofant/core/__init__.py
0.00s setup    diofant/core/cache.py
0.00s setup    diofant/core/tests/test_subs.py
0.00s setup    diofant/core/tests/test_evalf.py
0.00s setup    diofant/core/tests/test_cache.py
============================================= 67 passed, 1422 deselected in 103.55 seconds =============================================
```

Can you try this out with pytest 3.6.4 and pytest-pylint 0.11.0 for me?  I
recently released a change on how test collection works to be compatible
with pytest 3.7.0 and I'm wondering if I missed an edge case.

It would also be helpful to know if you are using a pylintrc file, and if
you have a pytest configuration setup either in pytest.ini, setup.cfg, or
tox.ini and what that looks like as well.

Thanks!

On Mon, Aug 6, 2018 at 11:26 AM, Sergey B Kirpichev <
notifications@github.com> wrote:

> This problem happens to me in the Diofant project (see all relevant
> configs here
> <https://github.com/diofant/diofant/tree/b50066efea3974cb0249c6bed6cab3fae9dc37c2>
> ).
>
> Consider adding a trailing space:
>
> $ git diff diofant/core/
> diff --git a/diofant/core/expr.py b/diofant/core/expr.py
> index c7857e0a5a..9fae5c3793 100644
> --- a/diofant/core/expr.py
> +++ b/diofant/core/expr.py
> @@ -30,7 +30,7 @@ class Expr(Basic, EvalfMixin, metaclass=ManagedProperties):
>
>      def __new__(cls, *args):
>          obj = Basic.__new__(cls, *args)
> -        obj._assumptions = cls.default_assumptions
> +        obj._assumptions = cls.default_assumptions
>          return obj
>
>      @property
>
> Now run py.test with this file selected:
>
>
> rootdir: /home/sk/src/diofant, inifile: setup.cfg
> plugins: xdist-1.22.5, timeout-1.3.1, pylint-0.12.1, forked-0.2, cov-2.5.1, hypothesis-3.66.23
> timeout: 600.0s
> timeout method: signal
> timeout func_only: False
> collected 34 items / 33 deselected
> -----------------------------------------------------------------
> Linting files
> .
> -----------------------------------------------------------------
>
> diofant/core/expr.py F                                                                                                           [100%]
>
> =============================================================== FAILURES ===============================================================
> ____________________________________________________ [pylint] diofant/core/expr.py _____________________________________________________
> C: 33,50: Trailing whitespace (trailing-whitespace)
> ====================================================== slowest 30 test durations =======================================================
> 0.00s setup    diofant/core/expr.py
> 0.00s call     diofant/core/expr.py
> 0.00s teardown diofant/core/expr.py
>
> Great, a problem detected.
>
> But what if we run py.test without arguments or using directory instead
> of file? Unfortunately, there will be no error in this case. For example:
>
> $ py.test -n0 --pylint -m pylint diofant/core
> ========================================================= test session starts ==========================================================
> platform linux -- Python 3.5.3, pytest-3.7.1, py-1.5.4, pluggy-0.7.1
>
> cache: True
> ground types: gmpy
>
> rootdir: /home/sk/src/diofant, inifile: setup.cfg
> plugins: xdist-1.22.5, timeout-1.3.1, pylint-0.12.1, forked-0.2, cov-2.5.1, hypothesis-3.66.23
> timeout: 600.0s
> timeout method: signal
> timeout func_only: False
> collected 1489 items / 1422 deselected
> -----------------------------------------------------------------
> Linting files
> .
> -----------------------------------------------------------------
>
> diofant/core/__init__.py .                                                                                                       [  1%]
> diofant/core/add.py .                                                                                                            [  2%]
> diofant/core/alphabets.py .                                                                                                      [  4%]
> diofant/core/assumptions.py .                                                                                                    [  5%]
> diofant/core/basic.py .                                                                                                          [  7%]
> diofant/core/cache.py .                                                                                                          [  8%]
> diofant/core/compatibility.py .                                                                                                  [ 10%]
> diofant/core/containers.py .                                                                                                     [ 11%]
> diofant/core/coreerrors.py .                                                                                                     [ 13%]
> diofant/core/decorators.py .                                                                                                     [ 14%]
> diofant/core/evalf.py .                                                                                                          [ 16%]
> diofant/core/evaluate.py .                                                                                                       [ 17%]
> diofant/core/expr.py .                                                                                                           [ 19%]
> diofant/core/exprtools.py .                                                                                                      [ 20%]
> diofant/core/facts.py .                                                                                                          [ 22%]
> diofant/core/function.py .                                                                                                       [ 23%]
> diofant/core/logic.py .                                                                                                          [ 25%]
> diofant/core/mod.py .                                                                                                            [ 26%]
> diofant/core/mul.py .                                                                                                            [ 28%]
> diofant/core/multidimensional.py .                                                                                               [ 29%]
> diofant/core/numbers.py .                                                                                                        [ 31%]
> diofant/core/operations.py .                                                                                                     [ 32%]
> diofant/core/power.py .                                                                                                          [ 34%]
> diofant/core/relational.py .                                                                                                     [ 35%]
> diofant/core/rules.py .                                                                                                          [ 37%]
> diofant/core/singleton.py .                                                                                                      [ 38%]
> diofant/core/strategies.py .                                                                                                     [ 40%]
> diofant/core/symbol.py .                                                                                                         [ 41%]
> diofant/core/sympify.py .                                                                                                        [ 43%]
> diofant/core/trace.py .                                                                                                          [ 44%]
> diofant/core/tests/__init__.py .                                                                                                 [ 46%]
> diofant/core/tests/test_args.py .                                                                                                [ 47%]
> diofant/core/tests/test_arit.py .                                                                                                [ 49%]
> diofant/core/tests/test_assumptions.py .                                                                                         [ 50%]
> diofant/core/tests/test_basic.py .                                                                                               [ 52%]
> diofant/core/tests/test_cache.py .                                                                                               [ 53%]
> diofant/core/tests/test_compatibility.py .                                                                                       [ 55%]
> diofant/core/tests/test_complex.py .                                                                                             [ 56%]
> diofant/core/tests/test_containers.py .                                                                                          [ 58%]
> diofant/core/tests/test_count_ops.py .                                                                                           [ 59%]
> diofant/core/tests/test_diff.py .                                                                                                [ 61%]
> diofant/core/tests/test_equal.py .                                                                                               [ 62%]
> diofant/core/tests/test_eval.py .                                                                                                [ 64%]
> diofant/core/tests/test_eval_power.py .                                                                                          [ 65%]
> diofant/core/tests/test_evalf.py .                                                                                               [ 67%]
> diofant/core/tests/test_evaluate.py .                                                                                            [ 68%]
> diofant/core/tests/test_expand.py .                                                                                              [ 70%]
> diofant/core/tests/test_expr.py .                                                                                                [ 71%]
> diofant/core/tests/test_exprtools.py .                                                                                           [ 73%]
> diofant/core/tests/test_facts.py .                                                                                               [ 74%]
> diofant/core/tests/test_function.py .                                                                                            [ 76%]
> diofant/core/tests/test_logic.py .                                                                                               [ 77%]
> diofant/core/tests/test_match.py .                                                                                               [ 79%]
> diofant/core/tests/test_multidimensional.py .                                                                                    [ 80%]
> diofant/core/tests/test_noncommutative.py .                                                                                      [ 82%]
> diofant/core/tests/test_numbers.py .                                                                                             [ 83%]
> diofant/core/tests/test_operations.py .                                                                                          [ 85%]
> diofant/core/tests/test_priority.py .                                                                                            [ 86%]
> diofant/core/tests/test_relational.py .                                                                                          [ 88%]
> diofant/core/tests/test_rules.py .                                                                                               [ 89%]
> diofant/core/tests/test_strategies.py .                                                                                          [ 91%]
> diofant/core/tests/test_subs.py .                                                                                                [ 92%]
> diofant/core/tests/test_symbol.py .                                                                                              [ 94%]
> diofant/core/tests/test_sympify.py .                                                                                             [ 95%]
> diofant/core/tests/test_trace.py .                                                                                               [ 97%]
> diofant/core/tests/test_truediv.py .                                                                                             [ 98%]
> diofant/core/tests/test_var.py .                                                                                                 [100%]
>
> ====================================================== slowest 30 test durations =======================================================
> 0.01s call     diofant/core/evalf.py
> 0.00s teardown diofant/core/expr.py
> 0.00s teardown diofant/core/coreerrors.py
> 0.00s setup    diofant/core/alphabets.py
> 0.00s setup    diofant/core/tests/test_args.py
> 0.00s setup    diofant/core/expr.py
> 0.00s setup    diofant/core/coreerrors.py
> 0.00s setup    diofant/core/add.py
> 0.00s setup    diofant/core/__init__.py
> 0.00s setup    diofant/core/tests/test_equal.py
> 0.00s setup    diofant/core/tests/test_expand.py
> 0.00s call     diofant/core/strategies.py
> 0.00s teardown diofant/core/tests/test_subs.py
> 0.00s teardown diofant/core/tests/test_operations.py
> 0.00s call     diofant/core/tests/test_truediv.py
> 0.00s call     diofant/core/tests/test_exprtools.py
> 0.00s setup    diofant/core/tests/test_numbers.py
> 0.00s call     diofant/core/tests/test_eval_power.py
> 0.00s call     diofant/core/tests/test_multidimensional.py
> 0.00s setup    diofant/core/tests/test_evaluate.py
> 0.00s setup    diofant/core/tests/test_function.py
> 0.00s teardown diofant/core/exprtools.py
> 0.00s teardown diofant/core/logic.py
> 0.00s setup    diofant/core/compatibility.py
> 0.00s setup    diofant/core/tests/test_noncommutative.py
> 0.00s call     diofant/core/__init__.py
> 0.00s setup    diofant/core/cache.py
> 0.00s setup    diofant/core/tests/test_subs.py
> 0.00s setup    diofant/core/tests/test_evalf.py
> 0.00s setup    diofant/core/tests/test_cache.py
> ============================================= 67 passed, 1422 deselected in 103.55 seconds =============================================
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/carsongee/pytest-pylint/issues/62>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAkYPX-VPGvmE_RDk1OYhz8SkD5RZjDFks5uOGAygaJpZM4VwmWA>
> .
>

> Can you try this out with pytest 3.6.4 and pytest-pylint 0.11.0 for me?

Hmm, that does work.

```
$ py.test -n0 --pylint -m pylint diofant/core
========================================================= test session starts ==========================================================
platform linux -- Python 3.5.3, pytest-3.6.4, py-1.5.4, pluggy-0.7.1

cache: True
ground types: gmpy

rootdir: /home/sk/src/diofant, inifile: setup.cfg
plugins: xdist-1.22.5, timeout-1.3.1, pylint-0.11.0, forked-0.2, cov-2.5.1, hypothesis-3.66.23
timeout: 600.0s
timeout method: signal
timeout func_only: False
collected 1489 items / 1422 deselected                                                                                                 
-----------------------------------------------------------------
Linting files
-----------------------------------------------------------------

diofant/core/__init__.py .                                                                                                       [  1%]
diofant/core/add.py .                                                                                                            [  2%]
diofant/core/alphabets.py .                                                                                                      [  4%]
diofant/core/assumptions.py .                                                                                                    [  5%]
diofant/core/basic.py .                                                                                                          [  7%]
diofant/core/cache.py .                                                                                                          [  8%]
diofant/core/compatibility.py .                                                                                                  [ 10%]
diofant/core/containers.py .                                                                                                     [ 11%]
diofant/core/coreerrors.py .                                                                                                     [ 13%]
diofant/core/decorators.py .                                                                                                     [ 14%]
diofant/core/evalf.py .                                                                                                          [ 16%]
diofant/core/evaluate.py .                                                                                                       [ 17%]
diofant/core/expr.py F                                                                                                           [ 19%]
diofant/core/exprtools.py .                                                                                                      [ 20%]
diofant/core/facts.py .                                                                                                          [ 22%]
diofant/core/function.py .                                                                                                       [ 23%]
diofant/core/logic.py .                                                                                                          [ 25%]
diofant/core/mod.py .                                                                                                            [ 26%]
diofant/core/mul.py .                                                                                                            [ 28%]
diofant/core/multidimensional.py .                                                                                               [ 29%]
diofant/core/numbers.py .                                                                                                        [ 31%]
diofant/core/operations.py .                                                                                                     [ 32%]
diofant/core/power.py .                                                                                                          [ 34%]
diofant/core/relational.py .                                                                                                     [ 35%]
diofant/core/rules.py .                                                                                                          [ 37%]
diofant/core/singleton.py .                                                                                                      [ 38%]
diofant/core/strategies.py .                                                                                                     [ 40%]
diofant/core/symbol.py .                                                                                                         [ 41%]
diofant/core/sympify.py .                                                                                                        [ 43%]
diofant/core/trace.py .                                                                                                          [ 44%]
diofant/core/tests/__init__.py .                                                                                                 [ 46%]
diofant/core/tests/test_args.py .                                                                                                [ 47%]
diofant/core/tests/test_arit.py .                                                                                                [ 49%]
diofant/core/tests/test_assumptions.py .                                                                                         [ 50%]
diofant/core/tests/test_basic.py .                                                                                               [ 52%]
diofant/core/tests/test_cache.py .                                                                                               [ 53%]
diofant/core/tests/test_compatibility.py .                                                                                       [ 55%]
diofant/core/tests/test_complex.py .                                                                                             [ 56%]
diofant/core/tests/test_containers.py .                                                                                          [ 58%]
diofant/core/tests/test_count_ops.py .                                                                                           [ 59%]
diofant/core/tests/test_diff.py .                                                                                                [ 61%]
diofant/core/tests/test_equal.py .                                                                                               [ 62%]
diofant/core/tests/test_eval.py .                                                                                                [ 64%]
diofant/core/tests/test_eval_power.py .                                                                                          [ 65%]
diofant/core/tests/test_evalf.py .                                                                                               [ 67%]
diofant/core/tests/test_evaluate.py .                                                                                            [ 68%]
diofant/core/tests/test_expand.py .                                                                                              [ 70%]
diofant/core/tests/test_expr.py .                                                                                                [ 71%]
diofant/core/tests/test_exprtools.py .                                                                                           [ 73%]
diofant/core/tests/test_facts.py .                                                                                               [ 74%]
diofant/core/tests/test_function.py .                                                                                            [ 76%]
diofant/core/tests/test_logic.py .                                                                                               [ 77%]
diofant/core/tests/test_match.py .                                                                                               [ 79%]
diofant/core/tests/test_multidimensional.py .                                                                                    [ 80%]
diofant/core/tests/test_noncommutative.py .                                                                                      [ 82%]
diofant/core/tests/test_numbers.py .                                                                                             [ 83%]
diofant/core/tests/test_operations.py .                                                                                          [ 85%]
diofant/core/tests/test_priority.py .                                                                                            [ 86%]
diofant/core/tests/test_relational.py .                                                                                          [ 88%]
diofant/core/tests/test_rules.py .                                                                                               [ 89%]
diofant/core/tests/test_strategies.py .                                                                                          [ 91%]
diofant/core/tests/test_subs.py .                                                                                                [ 92%]
diofant/core/tests/test_symbol.py .                                                                                              [ 94%]
diofant/core/tests/test_sympify.py .                                                                                             [ 95%]
diofant/core/tests/test_trace.py .                                                                                               [ 97%]
diofant/core/tests/test_truediv.py .                                                                                             [ 98%]
diofant/core/tests/test_var.py .                                                                                                 [100%]

=============================================================== FAILURES ===============================================================
____________________________________________________ [pylint] diofant/core/expr.py _____________________________________________________
C: 33,50: Trailing whitespace (trailing-whitespace)
====================================================== slowest 30 test durations =======================================================
0.00s teardown diofant/core/decorators.py
0.00s setup    diofant/core/exprtools.py
0.00s setup    diofant/core/mul.py
0.00s call     diofant/core/tests/test_eval_power.py
0.00s teardown diofant/core/evaluate.py
0.00s setup    diofant/core/tests/test_arit.py
0.00s setup    diofant/core/basic.py
0.00s setup    diofant/core/__init__.py
0.00s teardown diofant/core/tests/test_multidimensional.py
0.00s setup    diofant/core/alphabets.py
0.00s call     diofant/core/evalf.py
0.00s setup    diofant/core/tests/test_truediv.py
0.00s teardown diofant/core/rules.py
0.00s setup    diofant/core/tests/test_evaluate.py
0.00s setup    diofant/core/multidimensional.py
0.00s teardown diofant/core/tests/test_containers.py
0.00s call     diofant/core/mod.py
0.00s teardown diofant/core/tests/test_relational.py
0.00s teardown diofant/core/mod.py
0.00s setup    diofant/core/tests/test_var.py
0.00s teardown diofant/core/tests/test_priority.py
0.00s teardown diofant/core/tests/test_compatibility.py
0.00s call     diofant/core/expr.py
0.00s setup    diofant/core/tests/test_sympify.py
0.00s call     diofant/core/tests/test_evalf.py
0.00s setup    diofant/core/tests/test_basic.py
0.00s teardown diofant/core/tests/test_count_ops.py
0.00s setup    diofant/core/tests/test_expand.py
0.00s setup    diofant/core/singleton.py
0.00s call     diofant/core/relational.py
======================================== 1 failed, 66 passed, 1422 deselected in 79.17 seconds =========================================
```

> It would also be helpful to know if you are using a pylintrc file, and if you have a pytest configuration setup either in pytest.ini, setup.cfg

I have, but all relevant configs are in the referenced project tree (.pylintrc, setup.cfg and conftest.py).

We also are seeing the same issue.  I've got what looks like a minimal reproduction test case at https://github.com/jamur2/pytest-repro.

With current pytest and pytest-pylint, it only detects one failure:

```
± pipenv run pytest
============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.7.0, pytest-3.7.4, py-1.6.0, pluggy-0.7.1
rootdir: /Users/jackie.murphy/rally/pylint-repo, inifile: pytest.ini
plugins: pylint-0.12.1
collected 2 items
-----------------------------------------------------------------
Linting files
..
-----------------------------------------------------------------

myrepo/__init__.py F                                                                                                                                                       [ 50%]
myrepo/mymodule/__init__.py .                                                                                                                                              [100%]

==================================================================================== FAILURES ====================================================================================
__________________________________________________________________________ [pylint] myrepo/__init__.py ___________________________________________________________________________
C:  1,30: Trailing whitespace (trailing-whitespace)
======================================================================= 1 failed, 1 passed in 0.14 seconds =======================================================================
```

With pytest 3.6.4 and pytest-pylint 0.11.0, it correctly detects both failures:

```
± pipenv run pytest
============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.7.0, pytest-3.6.4, py-1.6.0, pluggy-0.7.1
rootdir: /Users/jackie.murphy/rally/pylint-repro-oldversion, inifile: pytest.ini
plugins: pylint-0.11.0
collected 2 items
-----------------------------------------------------------------
Linting files
-----------------------------------------------------------------

myrepo/__init__.py F                                                                                                                                                       [ 50%]
myrepo/mymodule/__init__.py F                                                                                                                                              [100%]

==================================================================================== FAILURES ====================================================================================
__________________________________________________________________________ [pylint] myrepo/__init__.py ___________________________________________________________________________
C:  1,30: Trailing whitespace (trailing-whitespace)
______________________________________________________________________ [pylint] myrepo/mymodule/__init__.py ______________________________________________________________________
C:  1,30: Trailing whitespace (trailing-whitespace)
============================================================================ 2 failed in 0.14 seconds ============================================================================
```

It looks like it's only looking one directory deep, but not recursing through subdirectories?

--- a/pytest_pylint.py
+++ b/pytest_pylint.py
@@ -202,7 +202,10 @@ class PyLintItem(pytest.Item, pytest.File):
         super(PyLintItem, self).__init__(fspath, parent)
 
         self.add_marker("pylint")
-        self.rel_path = get_rel_path(fspath.strpath, parent.fspath.strpath)
+        self.rel_path = get_rel_path(
+            fspath.strpath,
+            parent.session.fspath.strpath
+        )
 
         if msg_format is None:
             self._msg_format = '{C}:{line:3d},{column:2d}: {msg} ({symbol})'

